src/ui/game/player.rs

```diff
@@ -0,0 +1,412 @@
+use bevy::{
+    asset::{AssetServer, Handle},
+    ecs::{
+        component::Component,
+        entity::Entity,
+        query::{Changed, With},
+        system::{Commands, ParamSet, Query},
+    },
+    hierarchy::{BuildChildren, ChildBuilder, Children, DespawnRecursiveExt},
+    render::{color::Color, texture::Image},
+    ui::{
+        node_bundles::{ImageBundle, NodeBundle},
+        FlexDirection, Style, UiImage, UiRect, Val,
+    },
+    utils::default,
+};
+use thetawave_interface::{
+    abilities::{AbilityCooldownComponent, AbilitySlotIDComponent},
+    character::Character,
+    health::HealthComponent,
+    player::{self, PlayerComponent, PlayerIDComponent, PlayersResource},
+};
+
+use crate::{assets::UiAssets, player::CharactersResource};
+
+use super::parent::PlayerUiChildBuilderExt;
+
+const INNER_PADDING: UiRect = UiRect::all(Val::Percent(5.0));
+const INNER_WIDTH: Val = Val::Percent(35.0);
+const OUTER_PADDING: UiRect = UiRect::all(Val::Percent(5.0));
+const OUTER_WIDTH: Val = Val::Percent(65.0);
+const HEALTH_HEIGHT: Val = Val::Percent(55.0);
+const HEALTH_COLOR: Color = Color::CRIMSON;
+const HEALTH_EMPTY_ALPHA: f32 = 0.05;
+const HEALTH_FILLED_ALPHA: f32 = 0.75;
+const SHIELDS_HEIGHT: Val = Val::Percent(25.0);
+const SHIELDS_COLOR: Color = Color::CYAN;
+const SHIELDS_EMPTY_ALPHA: f32 = 0.05;
+const SHIELDS_FILLED_ALPHA: f32 = 0.75;
+const ARMOR_HEIGHT: Val = Val::Percent(20.0);
+const ARMOR_PADDING: UiRect = UiRect {
+    left: Val::Percent(0.0),
+    right: Val::Percent(0.0),
+    top: Val::Vh(0.1),
+    bottom: Val::Vh(0.1),
+};
+const ARMOR_COUNTER_ASPECT_RATIO: Option<f32> = Some(10.0);
+const ARMOR_COUNTER_MARGIN: UiRect =
+    UiRect::new(Val::Px(0.0), Val::Px(0.0), Val::Vh(0.1), Val::Vh(0.1));
+const ARMOR_COUNTER_COLOR: Color = Color::GOLD;
+const ARMOR_COUNTER_ALPHA: f32 = 0.75;
+const ABILITY_VALUE_COLOR: Color = Color::rgba(0.0, 0.0, 0.0, 0.85);
+
+// Player data Uis
+#[derive(Component)]
+pub struct HealthUi;
+
+#[derive(Component)]
+pub struct HealthValueUi;
+
+#[derive(Component)]
+pub struct ShieldsUi;
+
+#[derive(Component)]
+pub struct ShieldsValueUi;
+
+#[derive(Component)]
+pub struct ArmorUi;
+
+#[derive(Component)]
+pub struct ArmorCounterUi;
+
+#[derive(Component)]
+pub struct AbilitySlotUi;
+
+#[derive(Component)]
+pub struct AbilityIconUi;
+
+#[derive(Component)]
+pub struct AbilityValueUi;
+
+#[derive(Component)]
+pub struct PlayerUi;
+
+#[derive(Component)]
+pub struct PlayerInnerUi;
```

Comment:
New rule: no more just slapping `pub` on everything.

There is:

- `fn` the default. when the thing is just an implementation detail. 
- `pub(super)` visible to the parent module. This should be fairly common when you do submodules. But for the most part, each plugin needs this.

We probably dont need pub(crate)
but just consider why everything in the project needs to know about these things. If there is too much tangling, and if everything needs to be visible, then it might not be organized well.